contracts/incentives/base/BaseIncentivesController.sol

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.7.5;
+pragma experimental ABIEncoderV2;
+
+import {SafeMath} from '../../lib/SafeMath.sol';
+import {DistributionTypes} from '../../lib/DistributionTypes.sol';
+import {VersionedInitializable} from '@aave/aave-stake/contracts/utils/VersionedInitializable.sol';
+import {DistributionManager} from './DistributionManager.sol';
+import {IERC20} from '@aave/aave-stake/contracts/interfaces/IERC20.sol';
+import {IScaledBalanceToken} from '../../interfaces/IScaledBalanceToken.sol';
+import {IAaveIncentivesController} from '../../interfaces/IAaveIncentivesController.sol';
+
+/**
+ * @title BaseIncentivesController
+ * @notice Abstract contract template to build Distributors contracts for ERC20 rewards to protocol participants
+ * @author Aave
+ **/
+abstract contract BaseIncentivesController is
+  IAaveIncentivesController,
+  VersionedInitializable,
+  DistributionManager
+{
+  using SafeMath for uint256;
+
+  uint256 public constant REVISION = 1;
+
+  address public immutable override REWARD_TOKEN;
+
+  mapping(address => uint256) internal _usersUnclaimedRewards;
+
+  // this mapping allows whitelisted addresses to claim on behalf of others
+  // useful for contracts that hold tokens to be rewarded but don't have any native logic to claim Liquidity Mining rewards
+  mapping(address => address) internal _authorizedClaimers;
+
+  modifier onlyAuthorizedClaimers(address claimer, address user) {
+    require(_authorizedClaimers[user] == claimer, 'CLAIMER_UNAUTHORIZED');
+    _;
+  }
+
+  constructor(IERC20 rewardToken, address emissionManager)
+    DistributionManager(emissionManager)
+  {
+    REWARD_TOKEN = address(rewardToken);
+  }
+
+  /// @inheritdoc IAaveIncentivesController
+  function configureAssets(address[] calldata assets, uint256[] calldata emissionsPerSecond)
+    external
+    override
+    onlyEmissionManager
+  {
+    require(assets.length == emissionsPerSecond.length, 'INVALID_CONFIGURATION');
+
+    DistributionTypes.AssetConfigInput[] memory assetsConfig =
+      new DistributionTypes.AssetConfigInput[](assets.length);
+
+    for (uint256 i = 0; i < assets.length; i++) {
+      assetsConfig[i].underlyingAsset = assets[i];
+      assetsConfig[i].emissionPerSecond = uint104(emissionsPerSecond[i]);
+
+      require(assetsConfig[i].emissionPerSecond == emissionsPerSecond[i], 'INVALID_CONFIGURATION');
```

Comment:
The overflow check is done differently than similar checks in the `DistributionManager`, could probably be more aligned. 

From `DistributionManager::_updateAssetStateInternal()`:
`require(uint104(newIndex) == newIndex, 'Index overflow');`

Reply:
Done, better to be consistent between contracts.